File: algorithms/299. Bulls and Cows/solution.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        A = sum(i == j for i, j in zip(secret, guess))
        c1 = collections.Counter(secret)
        c2 = collections.Counter(guess)
        B = sum((c1 & c2).values())
        return '%dA%dB' % (A, B - A)

File: algorithms/299. Bulls and Cows/solution.cc
class Solution {
public:
    string getHint(string secret, string guess) {
        int A = 0, B = 0;
        int count[10] = {0};
        for ( int i = 0; i < secret.size(); ++i ) {
            int s = secret[i] - '0';
            int g = guess[i] - '0';
            if ( s == g ) {
                ++A;
                continue;
            }
            if ( count[s] < 0 ) ++B;
            if ( count[g] > 0 ) ++B;
            ++count[s];
            --count[g];
        }
        return to_string(A) + "A" + to_string(B) + "B";
    }
};
